coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -3351,6 +3351,20 @@ def _internal_op_tensor_inplace_fill(context, node):
     data = context[node.inputs[0]]
     fill_scalar = context[node.inputs[1]]
 
+    if len(node.inputs) == 2 and not any_symbolic(fill_scalar.val):
+        shape = mb.shape(x=data)
+        if isinstance(fill_scalar.val, _np.ndarray):
+            np_dtype = fill_scalar.val.dtype
+            if _np.issubdtype(np_dtype, _np.integer):
+                fill_scalar = int(fill_scalar.val)
+            elif _np.issubdtype(np_dtype, _np.floating):
+                fill_scalar = float(fill_scalar.val)
+            else:
+                raise ValueError(f"Unsupported op {node} with target dtype {np_dtype}")
```

Comment:
I fee like the return dtype is determined by the dtype of the source tensor not the filled value?
```
>>> a = torch.tensor([[2,3],[4,5]])
>>> a
tensor([[2, 3],
        [4, 5]])
>>> a.fill_(4)
tensor([[4, 4],
        [4, 4]])
>>> a = torch.tensor([[2.,3.],[4.,5.]])
>>> a.fill_(4)
tensor([[4., 4.],
        [4., 4.]])
>>> a = torch.tensor([[2,3],[4,5]])
>>> a.fill_(4.5)
tensor([[4, 4],
        [4, 4]])
```
Can you please check it?